Pixel Game/Assets/Scripts/InventorySlot.cs

```diff
@@ -70,6 +76,17 @@ public void UseItem()
                     ClearSlot();
                 }
             }
+            // Using a Quest item when in range of the target delivery npc will mark the quest as finished
+            // and destroy said quest item
+            else if(ItemDataInSlot.itemType == ItemData.ITEMTYPE.QUEST_ITEM)
+            {
+                if (dataToPass.currentActivateNpc.Equals(dataToPass.currentActivePlayerQuest.npcDeliveryTarget.name))

```

Comment:
I thought when comparing string i should use .Equals()
Is it wrong? or a better question. What is the better choice? :)

Reply:
You could do
```c#
if (myString1 == myString2)
```
In Java this will check if the variables are pointing to the same string object, so there you have to use Equals to compare the actual content.
I had to google for difference between == and Equals in C#. Apparently it's the same as in Java except when it comes to strings, where there is no difference. Just that == looks better IMO.
https://www.c-sharpcorner.com/UploadFile/3d39b4/difference-between-operator-and-equals-method-in-C-Sharp/

Reply:
Okay got it! :) lets stick to ==